skills/src/main/java/cm/aptoide/skills/usecase/GetTopUpListUseCase.kt

```diff
@@ -0,0 +1,36 @@
+package cm.aptoide.skills.usecase
+
+import cm.aptoide.skills.interfaces.WalletAddressObtainer
+import cm.aptoide.skills.model.Gateway
+import cm.aptoide.skills.model.TopUpStatus
+import cm.aptoide.skills.model.TransactionType
+import cm.aptoide.skills.repository.TopUpRepository
+import io.reactivex.Single
+import javax.inject.Inject
+
+class GetTopUpListUseCase @Inject constructor(
+  private val topUpRepository: TopUpRepository,
+  private val walletAddressObtainer: WalletAddressObtainer
+) {
+  operator fun invoke(type: TransactionType, status: TopUpStatus): Single<Status> {
+    return walletAddressObtainer.getWalletAddress()
+      .flatMap { wallet ->
+        topUpRepository.getTopUpHistory(type, status, wallet.address)
+          .toObservable()
+          .flatMapIterable { it.items }
+          .filter { it.gateway?.name != Gateway.Name.myappcoins }
+          .toList().map { transaction ->
+            if(transaction.isEmpty()){
+              Status.NO_TOPUP
+            }
+            if (transaction.any{it.gateway?.name == Gateway.Name.myappcoins} ){
+              Status.PAYMENT_METHOD_NOT_SUPPORTED
+            }
+            Status.AVAILABLE
```

Comment:
I would change this to a when instead of a chained if, even tho github doesnt allow the right format, it would be something like this
` when {`
`    transaction.isEmpty() -> Status.NO_TOPUP`
`    transaction.any { it.gateway?.name == Gateway.Name.myappcoins } -> Status.PAYMENT_METHOD_NOT_SUPPORTED`
`    else -> Status.AVAILABLE`
` }`
   
Also, the Status.PAYMENT_METHOD_NOT_SUPPORTED would never happen because of the filter before the last map.
In order to map it this way, we cant filter out that option